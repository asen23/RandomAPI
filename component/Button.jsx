export default function Button(props) {
    return (
        <button onClick={props.onClick} className="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white">
            {props.children}
        </button>
    )
}