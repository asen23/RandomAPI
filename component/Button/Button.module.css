.button {
    border: 1px solid black;
    border-radius: 3px;
    background-color: antiquewhite;
}

.button:hover {
    background-color: rgb(247, 206, 152);
}
