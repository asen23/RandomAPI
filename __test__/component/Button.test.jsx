import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Button from "component/Button"

describe("Button", () => {
    it("renders unchanged", async () => {
        const { asFragment } = render(<Button>A Component</Button>)
        expect(asFragment()).toMatchSnapshot()
    })

    it("renders its children", () => {
        render(<Button>A Component</Button>)

        const child = screen.getByText("A Component")
        expect(child).toBeInTheDocument()
    })

    it("calls its onclick", () => {
        const onClick = jest.fn()
        render(<Button onClick={onClick}>A Component</Button>)
    
        const button = screen.getByRole("button")
        userEvent.click(button)

        expect(onClick.mock.calls.length).toBe(1)
    })
})
