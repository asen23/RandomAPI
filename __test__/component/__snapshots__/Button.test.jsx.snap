// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders unchanged 1`] = `
<DocumentFragment>
  <button
    class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
  >
    A Component
  </button>
</DocumentFragment>
`;
